components/logins/src/encryption.rs

```diff
@@ -28,22 +28,82 @@
 // To make life a little easier, we do that via a struct.
 
 use crate::error::*;
+use std::sync::Arc;
 
-pub type EncryptorDecryptor = jwcrypto::EncryptorDecryptor<Error>;
+pub trait EncryptorDecryptor: Send + Sync {
+    fn encrypt(&self, cleartext: Vec<u8>, description: String) -> ApiResult<Vec<u8>>;
+    fn decrypt(&self, ciphertext: Vec<u8>, description: String) -> ApiResult<Vec<u8>>;
+}
+
+pub trait KeyManager: Send + Sync {
+    fn get_key(&self) -> ApiResult<Vec<u8>>;
+}
+
+pub struct ManagedEncryptorDecryptor {
+    key_manager: Arc<dyn KeyManager>,
+}
+
+impl ManagedEncryptorDecryptor {
+    pub fn new(key_manager: Arc<dyn KeyManager>) -> Self {
+        Self { key_manager }
+    }
+}
+
+impl EncryptorDecryptor for ManagedEncryptorDecryptor {
+    #[handle_error(Error)]
+    fn encrypt(&self, cleartext: Vec<u8>, description: String) -> ApiResult<Vec<u8>> {
+        let key = self
+            .key_manager
+            .get_key()
+            .map_err(|_| Error::EncryptionKeyMissing)?;
+
+        let encdec = jwcrypto::EncryptorDecryptor::new(std::str::from_utf8(&key)?)?;
+        encdec
+            .encrypt(std::str::from_utf8(&cleartext)?, &description)
+            .map(|text| text.into())
+    }
+
+    #[handle_error(Error)]
+    fn decrypt(&self, ciphertext: Vec<u8>, description: String) -> ApiResult<Vec<u8>> {
+        let key = self
+            .key_manager
+            .get_key()
+            .map_err(|_| Error::EncryptionKeyMissing)?;
+        let encdec = jwcrypto::EncryptorDecryptor::new(std::str::from_utf8(&key)?)?;
+        encdec
+            .decrypt(std::str::from_utf8(&ciphertext)?, &description)
+            .map(|text| text.into())
+    }
+}
+
+pub struct StaticKeyManager {
+    pub key: String,
+}
+impl StaticKeyManager {
+    pub fn new(key: String) -> Self {
+        Self { key }
+    }
+}
+impl KeyManager for StaticKeyManager {
+    #[handle_error(Error)]
+    fn get_key(&self) -> ApiResult<Vec<u8>> {
+        Ok(self.key.as_bytes().into())
+    }
+}
```

Comment:
IIUC consumers have 3 options for this:
  - Implement `EncryptorDecryptor` directly
  - Implement `KeyManager`
  - Generate a single key and create a `StaticKeyManager`

If so, I like the design a lot.  We should probably document it at some point, but that can be a follow-up.

Reply:
Yes, thats how it's intended.

```rust
// Implement EncryptorDecryptor directly:
LoginStore::new(MyEncryptorDecryptor)
// Implement KeyManager:
LoginStore::new(ManagedEncryptorDecryptor::new(MyKeyManager))
// Generate a single key and create a StaticKeyManager:
LoginStore::new(ManagedEncryptorDecryptor::new(StaticKeyManager { key: myKey }))
```
(simplified, w/o the use of Arcs)

Is this the design you expected?

I will add and update the documentation within this PR.

Reply:
done